test/es-module/test-typescript-eval.mjs

```diff
@@ -4,7 +4,6 @@ import { test } from 'node:test';
 
 test('eval TypeScript ESM syntax', async () => {
   const result = await spawnPromisified(process.execPath, [
-    '--input-type=module',
```

Comment:
It seems we should keep the original test no? IMO it's best to duplicate it, so if it starts failing we can know whether it's the module detection or something else that's broken

Reply:
The weird behavior was caused by putting default type module, but I guess yeah we can duplicate

Reply:
done